src/components/RedirectionNotificationModal.js

```diff
@@ -1,72 +1,160 @@
 import Modal from "@mui/material/Modal";
+import { Link } from "react-router-dom/cjs/react-router-dom";
 import CloseRoundedIcon from "@mui/icons-material/CloseRounded";
 import * as s from "../components/RedirectionNotificationModal.sc";
+import { isMobile } from "../utils/misc/browserDetection";
 
 //This modal is used in the ArticlePreview component
 
 //TODO: Further refactor, e.g after testing and making sure users
 //understand text and wording, turn strings into variables and move to definitions.js
 
 export default function RedirectionNotificationModal({
+  api,
   article,
   open,
-  handleClose, //handleClose function defined in the ArticlePreview.js, passed as a prop
+  handleClose,
+  selectedDoNotShowRedirectionModal, //related to the "Do not show" checkbox selection
+  setSelectedDoNotShowRedirectionModal, //related to the "Do not show" checkbox selection
+  setOpenedExternallyWithoutModal, //related to the modal use based on the "Do not show" selection
+  setIsArticleSaved, // related to the article's state
 }) {
+  function handleVisibilityCheckboxSelection() {
```

Comment:
based on the implementation, the name for this should be `toggleRedirectionCheckboxSelection` maybe? toggle = change the state to the opposite, which is exactly what you do here.